pkg/kapp/clusterapply/delete_change.go

```diff
@@ -65,7 +65,7 @@ func (c DeleteChange) IsDoneApplying() (ctlresm.DoneApplyState, []string, error)
 		return ctlresm.DoneApplyState{}, nil, err
 	}
 
-	return ctlresm.DoneApplyState{Done: !exists, Successful: true}, nil, nil
+	return ctlresm.DoneApplyState{Done: !exists, Successful: true}, buildDescMsg(res, !exists), nil
```

Comment:
instead of this, it should be something like:

```
if exists {
  return ctlresm.DoneApplyState{Done: true, Successful: true}, nil, nil
}

return ctlresm.NewDeleting(res).IsDoneApplying(), nil, nil
```

make sure to double check that message shows up in correct place for both cases.

Reply:
@cppforlife , While doing `ctlresm.NewDeleting(existingRes).IsDoneApplying()` we need to check for `ctlresm.NewDeleting(existingRes)` is returning non nil. At this stage resource doesn't get marked as deleting and `ctlresm.NewDeleting(existingRes)` returns nil.

Reply:
@cppforlife  `ctlresm.NewDeleting(existingRes)` is returning nil for some resource and result in nil pointer reference. 

Observed when deploying a Service resource (without finalizer) then `Service` and its `Endpoint` resource get deleted and `exists` come `false` for them but `EndpointSlice` do exist and `deletionTimestamp` is not present in `existingRes` of `EndpointSlice` so we have to add the condition to check for `res.IsDeleting()` for such usecases. 

I have started doing this in `descMessage()`